include/deal.II/lac/block_matrix_base.h

```diff
@@ -103,7 +103,7 @@ namespace BlockMatrixIterators
 
     // Let the iterator class be a friend.
     template <typename>
-    friend class MatrixIterator;
+    friend class dealii::MatrixIterator;
```

Comment:
This declaration adds a tentative declaration of a class `MatrixIterator` to the current namespace. Any later actual declaration of a class with this name will then be merged with this tentative declaration.